src/store/modules/agreements.js

```diff
@@ -0,0 +1,58 @@
+import Vue from 'vue'
+import agreements from '@/services/api/agreements'
+
+export const types = {
+  REQUEST_ENTRY: 'Request Entry',
+  REQUEST_AGREE: 'Request Agree',
+  REQUEST_CREATE: 'Request Create',
+  REQUEST_SAVE: 'Request Save',
+  REQUEST_DESTROY: 'Request Destroy',
+  RECEIVE_ENTRY: 'Receive Entry',
+}
+
+function initialState () {
+  return {
+    entries: {},
+  }
+}
+
+export const state = initialState()
+
+export const getters = {
+  get: state => id => {
+    return state.entries[id]
+  },
+}
+
+export const actions = {
+  async fetch ({ commit }, id) {
+    commit(types.REQUEST_ENTRY)
+    commit(types.RECEIVE_ENTRY, { agreement: await agreements.get(id) })
+  },
+  async agree ({ commit }, id) {
+    commit(types.REQUEST_AGREE)
+    commit(types.RECEIVE_ENTRY, { agreement: await agreements.agree(id) })
+  },
+  async create ({ commit }, agreement) {
+    commit(types.REQUEST_CREATE)
+    agreement = await agreements.create(agreement)
```

Comment:
should handle server validation errors, maybe track the to do tasks in an issue?
by the way, I would rename the `onlyHandleAPIError` helper to `handleServerValidationError`, what do you think about it?